File: src/main/java/com/classy/instagram/article/entity/Article.java
package com.classy.instagram.article.entity;

import com.classy.instagram.user.entity.User;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import java.sql.Timestamp;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Article {
    @Column(name = "likes_count")
    @ColumnDefault("0")
    public int likesCount;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String content;
    @ManyToOne(fetch = FetchType.LAZY)
    private User author;
    private Timestamp createTime;
    private Timestamp updateTime;
    private String pictureUrl;
    @OneToMany(cascade = CascadeType.REMOVE, orphanRemoval = true)
    private List<Reply> reply;
    @OneToMany(cascade = CascadeType.REMOVE, orphanRemoval = true)
    private List<ArticleLike> like;
}

File: src/main/kotlin/com/classy/placelog/article/entity/Article.kt
package com.classy.placelog.article.entity

import com.classy.placelog.common.entity.BaseEntity
import com.classy.placelog.user.entity.User
import jakarta.persistence.Column
import jakarta.persistence.ElementCollection
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import jakarta.persistence.OrderColumn

@Entity
class Article(
    /** 게시글 ID */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,
    /** 글 제목 */
    @Column
    var title: String,
    /** 글 내용 */
    @Column(columnDefinition = "LONGTEXT")
    var content: String,
    /** 이미지 링크들 */
    @ElementCollection
    @OrderColumn
    var images: List<String>,
    /** 좋아요 수 */
    @Column(name = "like_count")
    var likeCount: Long = 0L,
    /** 댓글 수 */
    @Column(name = "reply_count")
    var replyCount: Long = 0L,
    /** 작성자 */
    @ManyToOne(fetch = FetchType.LAZY)
    val author: User
) : BaseEntity() {
}
